index.js

```diff
@@ -71,7 +87,11 @@ Collector.prototype.flush = function() {
 
     // only send N points at a time to avoid making requests too large
     // TODO what if we are backed up?
-    var points = self._points.splice(0, 50);
+
+    var spliceIndex = self.computePointCountToSend(self._points.map(function (point) {
+      return JSON.stringify(point).length;
```

Comment:
I assumed that this might be the case, however there's no flushing happening in `influx-udp` so part of the change would still have to live here. One way to apply your idea would be to expose a method from `influx-udp` that computes the size of the message on the wire and then use this here.

Is this what you had in mind?